crates/jstzd/src/task/jstzd.rs

```diff
@@ -0,0 +1,115 @@
+use super::{octez_node::OctezNode, Task};
+use anyhow::Result;
+use async_dropper_simple::AsyncDrop;
+use async_trait::async_trait;
+use octez::r#async::node_config::OctezNodeConfig;
+use tokio::task::JoinHandle;
+
+struct Jstzd {
+    octez_node: OctezNode,
+}
+
+#[derive(Clone)]
+pub struct JstzdConfig {
+    octez_node_config: OctezNodeConfig,
+}
+
+impl JstzdConfig {
+    pub fn new(octez_node_config: OctezNodeConfig) -> Self {
+        Self { octez_node_config }
+    }
+}
+
+#[async_trait]
+impl Task for Jstzd {
+    type Config = JstzdConfig;
+
+    async fn spawn(config: Self::Config) -> Result<Self> {
+        Ok(Self {
+            octez_node: OctezNode::spawn(config.octez_node_config.clone()).await?,
+        })
+    }
+
+    async fn kill(&mut self) -> Result<()> {
+        self.octez_node.kill().await
+    }
+
+    async fn health_check(&self) -> Result<bool> {
+        self.octez_node.health_check().await
+    }
+}
+
+pub struct JstzdServer {
+    jstzd: Option<Jstzd>,
+    jstzd_config: JstzdConfig,
+    jstzd_server_port: u16,
+    server_handle: Option<JoinHandle<()>>,
+}
+
+#[async_trait]
+impl AsyncDrop for JstzdServer {
+    async fn async_drop(&mut self) {
+        if let Err(e) = self.stop().await {
+            eprintln!("failed to stop jstzd server: {:?}", e);
+        }
+    }
+}
+
+impl JstzdServer {
+    pub fn new(config: JstzdConfig, port: u16) -> Self {
+        Self {
+            jstzd: None,
+            jstzd_config: config,
+            jstzd_server_port: port,
+            server_handle: None,
+        }
+    }
+
+    pub async fn health_check(&self) -> bool {
+        let res =
+            reqwest::get(format!("http://localhost:{}", self.jstzd_server_port)).await;
+        if res.is_err() {
+            return false;
+        }
```

Comment:
Making a request to itself is quite odd. Maybe check `server_handle`  instead?

Reply:
I think it's fine. I'm not quite sure about checking if `server_handle` is finished since it might be possible that somehow the server gets stuck but the task is still running, though the chance is relatively low since this server is pretty much serving static content only.

Reply:
Yeah, I reckon that would hardly happen on the jstzd server.

And I guess the `health_check` would be bound to something like `/health` which means that handling the endpoint proves that is isn't stuck?

Reply:
Alright. I've updated this check.